group/asg/main.tf

```diff
@@ -0,0 +1,54 @@
+# AWS Auto Scaling Group
+
+## Creates auto scaling group
+resource "aws_autoscaling_group" "asg" {
+  count                     = "${index(split(",",var.opposite),signum(length(var.min_elb_capacity)))}"
+  name                      = "${var.stack_item_label}-asg"
+  max_size                  = "${var.max_size}"
+  min_size                  = "${var.min_size}"
+  launch_configuration      = "${var.lc_id}"
+  health_check_grace_period = "${var.hc_grace_period}"
+  health_check_type         = "EC2"
+  force_delete              = "${var.force_delete}"
+  vpc_zone_identifier       = ["${split(",",var.subnets)}"]
+  wait_for_capacity_timeout = "${var.wait_for_capacity_timeout}"
+
+  tag {
+    key                 = "application"
+    value               = "${var.stack_item_fullname}"
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "Name"
+    value               = "${var.stack_item_label}"
+    propagate_at_launch = true
+  }
+}
+
+resource "aws_autoscaling_group" "asg_elb" {
+  count                     = "${signum(length(var.min_elb_capacity))}"
```

Comment:
This is evaluated based on whether the input is an empty string or not. A value of 0 would still trigger this properly.

I actually wanted to key this off of 'load_balancers', which seemed more logical. The problem I ran into though was that since the ELB name is calculated at runtime the math didn't work out.